### PR TITLE
[1.5.0] Updated doc for flags 

### PR DESCRIPTION
**Pull request by anushkamittal20**
Signed-off-by: anushkamittal20 <anumittal4641@gmail.com>

To make the flag names style consistent (https://github.com/kyverno/kyverno/issues/1991) all flag names except port names are being made camelCase in 1.5.0 (https://github.com/kyverno/kyverno/pull/2467). This PR reflects these changes in the documentation.

### CODE REVIEW

File: content/en/docs/Installation/_index.md [hunk 1]
```diff
@@ -338,14 +338,23 @@ kubectl logs -l app=kyverno -n <namespace>
...
 3. `filterK8sResources`(deprecated): Kubernetes resources in the format "[kind,namespace,name]" where the policy is not evaluated by the admission webhook. For example --filterKind "[Deployment, kyverno, kyverno]" --filterKind "[Deployment, kyverno, kyverno],[Events, *, *]".
-4. `gen-workers`: the number of workers for processing generate policies concurrently. Default is set to 10.
-5. `background-scan`: the interval (like 30s, 15m, 12h) for background processing. Default is set to 1h.
+4. `genWorkers`: the number of workers for processing generate policies concurrently. Default is set to 10.
+5. `backgroundScan`: the interval (like 30s, 15m, 12h) for background processing. Default is set to 1h.
 6. `generateSuccessEvents`: specifies whether (true/false) to generate success events. Default is set to "false".
+7. `kubeconfig`: specifies path to a kubeconfig. Only required if out-of-cluster.
+8. `serverIP`: specifies IP address where Kyverno controller runs. Only required if out-of-cluster.
+9. `profile`: setting this flag to 'true' will enable profiling.
+10. `profile-port`: specifies port to enable profiling at, defaults to 6060.
```
Comment (realshuting) [inline content/en/docs/Installation/_index.md:352]: Didn't we change it to `profilePort`?

File: content/en/docs/Installation/_index.md [hunk 2]
```diff
@@ -338,14 +338,23 @@ kubectl logs -l app=kyverno -n <namespace>
...
 3. `filterK8sResources`(deprecated): Kubernetes resources in the format "[kind,namespace,name]" where the policy is not evaluated by the admission webhook. For example --filterKind "[Deployment, kyverno, kyverno]" --filterKind "[Deployment, kyverno, kyverno],[Events, *, *]".
-4. `gen-workers`: the number of workers for processing generate policies concurrently. Default is set to 10.
-5. `background-scan`: the interval (like 30s, 15m, 12h) for background processing. Default is set to 1h.
+4. `genWorkers`: the number of workers for processing generate policies concurrently. Default is set to 10.
+5. `backgroundScan`: the interval (like 30s, 15m, 12h) for background processing. Default is set to 1h.
 6. `generateSuccessEvents`: specifies whether (true/false) to generate success events. Default is set to "false".
+7. `kubeconfig`: specifies path to a kubeconfig. Only required if out-of-cluster.
+8. `serverIP`: specifies IP address where Kyverno controller runs. Only required if out-of-cluster.
```
Comment (realshuting) [inline content/en/docs/Installation/_index.md:350]: Let's remove these 2 as they are used out-of-cluster.

File: content/en/docs/Installation/_index.md [hunk 3]
```diff
@@ -338,14 +338,23 @@ kubectl logs -l app=kyverno -n <namespace>
...
+9. `profile`: setting this flag to 'true' will enable profiling.
+10. `profile-port`: specifies port to enable profiling at, defaults to 6060.
+11. `disableMetrics`: specifies whether (true/false) to enable exposing the metrics. Default is set to "false".
+12. `metrics-port`: specifies the port to expose prometheus metrics, default to port 8000.
```
Comment (realshuting) [inline content/en/docs/Installation/_index.md:354]: `metricsPort`?

File: content/en/docs/Installation/_index.md (hunk 1 above)
Comment (anushkamittal20) [reply content/en/docs/Installation/_index.md:352]: In a conversation I read while reviewing the issue https://github.com/kyverno/kyverno/pull/2038#discussion_r663107111 I thought that the port names need to be hyphenated and not camelCase.
Do let me know if I should make any changes regarding this.

File: content/en/docs/Installation/_index.md (hunk 3 above)
Comment (anushkamittal20) [reply content/en/docs/Installation/_index.md:354]: I have mentioned the issue with camelCase for port names in a previous comment, it is that Kubernetes does not allow camel case for port names.

File: content/en/docs/Installation/_index.md [hunk 4]
```diff
@@ -340,12 +340,22 @@ kubectl logs -l app=kyverno -n <namespace>
 
 The following flags are used to control the behavior of Kyverno and must be set in the Kyverno ConfigMap.
```
Comment (realshuting) [inline content/en/docs/Installation/_index.md:341]: "...must be set in the Kyverno ConfigMap."

Are all these flags supported via ConfigMap?

Comment (realshuting) [reply content/en/docs/Installation/_index.md:341]: I don't think `genWorkers` is supported via ConfigMap, where did you see that?

**Review by chipzoller — COMMENTED** [review]
Need @realshuting input here.

**Comment by realshuting** [comment]
Sorry I missed this comment.

Here are all the container flags we support in 1.5.0, with Anushka's change, all the dash-separated flags were deprecated and replaced by camelCase flags.
https://github.com/kyverno/kyverno/blob/e5e849acfebd0461ae147d066821fdf86c620fc7/cmd/kyverno/main.go#L73-L98

File: content/en/docs/Installation/_index.md [hunk 5]
```diff
@@ -355,11 +355,16 @@ The following flags are used to control the behavior of Kyverno and must be set
 The following flags can also be used to control the advanced behavior of Kyverno and must be set on the main `kyverno` container in the form of arguments.
 
 1. `-v`: Sets the verbosity mode of Kyverno log output. Takes an integer from 1 to 6 with 6 being the most verbose.
-2. `--background-scan`: The interval (like 30s, 15m, 12h) for background processing resulting in policy report entries. Default is set to 1h.
-3. `--webhookTimeout`: Sets the timeout for a webhook response from Kyverno. Value is in seconds. This is deprecated in Kyverno 1.5.0.
-4. `--gen-workers`: the number of workers for processing generate policies concurrently. Default is set to 10.
-5. `--generateSuccessEvents`: specifies whether (true/false) to generate success events. Default is set to "false".
-6. `--autoUpdateWebhooks`: auto-configuration of the webhooks based on installed policies. Default is set to "true".
+2. `profile`: setting this flag to 'true' will enable profiling.
```
Comment (chipzoller) [inline content/en/docs/Installation/_index.md:358]: @anushkamittal20 I don't see "webhookTimeout" in this PR.

https://github.com/kyverno/kyverno/blob/e5e849acfebd0461ae147d066821fdf86c620fc7/cmd/kyverno/main.go#L73-L98

File: content/en/docs/Installation/_index.md [hunk 6]
```diff
@@ -355,11 +355,16 @@ The following flags are used to control the behavior of Kyverno and must be set
...
+3. `profilePort`: specifies port to enable profiling at, defaults to 6060.
+4. `metricsPort`: specifies the port to expose prometheus metrics, default to port 8000.
+5. `genWorkers`: the number of workers for processing generate policies concurrently. Default is set to 10.
+6. `disableMetrics`: specifies whether (true/false) to enable exposing the metrics. Default is set to "false".
+7. `backgroundScan`: the interval (like 30s, 15m, 12h) for background processing. Default is set to 1h.
+8. `imagePullSecrets`: specifies secret resource names for image registry access credentials.
+9. `autoUpdateWebhooks`: Set this flag to 'false' to disable auto-configuration of the webhook. Default is set to "true".
+10. `generateSuccessEvents`: specifies whether (true/false) to generate success events. Default is set to "false".
```
Comment (chipzoller) [inline content/en/docs/Installation/_index.md:366]: I don't see "generateSuccessEvents" in this code block pointed out by @realshuting 

https://github.com/kyverno/kyverno/blob/e5e849acfebd0461ae147d066821fdf86c620fc7/cmd/kyverno/main.go#L73-L98

Comment (chipzoller) [reply content/en/docs/Installation/_index.md:366]: Is it "deprecated" or is it "removed" in 1.5.0? Docs say "deprecated".

File: content/en/docs/Installation/_index.md (hunk 5 above)
Comment (chipzoller) [reply content/en/docs/Installation/_index.md:358]: Can you also add the `-v` flag, please? Even though this isn't Kyverno specific, users need to know this is a flag that can be passed to the main container to turn up verbose logging.

**Review by chipzoller — CHANGES_REQUESTED** [review]
Just one point on the "webhookTimeout" flag I'd like to understand here.

File: content/en/docs/Installation/_index.md (hunk 6 above)
Comment (realshuting) [reply content/en/docs/Installation/_index.md:366]: Sorry it should be ConfigMap variable:
https://github.com/kyverno/kyverno/blob/e5e849acfebd0461ae147d066821fdf86c620fc7/pkg/config/dynamicconfig.go#L41

Comment (realshuting) [reply content/en/docs/Installation/_index.md:366]: Where did you say it's deprecated?

File: content/en/docs/Installation/_index.md (hunk 5 above)
Comment (realshuting) [reply content/en/docs/Installation/_index.md:358]: `webhookTimeout ` has been deprecated, should use `spec.webhookTimeoutSeconds` instead.

**Comment by chipzoller** [comment]
@anushkamittal20 ?

**Comment by chipzoller** [comment]
@anushkamittal20 can you make these changes so this PR can be merged?

**Comment by anushkamittal20** [comment]
@chipzoller I am sorry I missed this, I will push these changes after college.

File: content/en/docs/Installation/_index.md (hunk 5 above)
Comment (anushkamittal20) [reply content/en/docs/Installation/_index.md:403]: Should I add something like 
```
--spec.webhookTimeoutSeconds: Sets the timeout for a webhook response from Kyverno. Value is in seconds. This is deprecated in Kyverno 1.5.0.
``` 
@chipzoller @realshuting

File: content/en/docs/Installation/_index.md (hunk 6 above)
Comment (anushkamittal20) [reply content/en/docs/Installation/_index.md]: I have moved `generateSuccessEvents`

File: content/en/docs/Installation/_index.md (hunk 5 above)
Comment (chipzoller) [reply content/en/docs/Installation/_index.md:404]: It's a field in a policy or webhook config `spec`, right, @realshuting ?

**Review by chipzoller — APPROVED** [review]

**Pull request merged**